### PR TITLE
feat(core): Add dynamic server URL

### LINKED ISSUE #419: `server` - replace logic of getting server URLs
The current architecture of the server contains dedicated network and db per server. Example, if we're on testnet, we will use "https://testnet.api.defichain.com (not the final url, just an example)". Same as we config WhaleContexts. We will need to modify https://github.com/WavesHQ/bridge/blob/main/apps/web/src/store/website.tsx to be able to support this. 

#### What would you like to be added:
- [ ] Replace logic of getting server URLs depending on selected network on the client. 
- [ ] Example, if network is "Local" use "http://localhost:5741", for others, will fill up once deployed
- [ ] Rename the file to "defichain" instead of website
- [ ] Use a similar concept as `newOceanOptions`

### PR DESCRIPTION
**Pull request by kevzzsk**
#### What this PR does / why we need it:

Add dynamic switch to API url based on the current network environment selected.

#### Which issue(s) does this PR fixes?:

Fixes #419 

#### Developer Checklist:

- [x] Read your code changes at least once
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode\*
- [ ] Your UI implementation visually matched the rendered design\*
- [ ] Unit tests\*
- [ ] Added e2e tests\*
- [ ] Added translations\*

### CODE REVIEW

File: apps/web/src/store/defichain.tsx [hunk 1]
```diff
@@ -39,15 +39,14 @@ export const bridgeApi = createApi({
     verify: builder.query<
       any,
       {
-        network: string;
+        baseUrl?: string;
```
Comment (kevzzsk) [inline apps/web/src/store/defichain.tsx:42]: not sure if this is the best way without TS screaming.

File: apps/web/src/config/networkUrl.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { EnvironmentNetwork } from "@waveshq/walletkit-core";
+
+const BASE_URLS: { [key in EnvironmentNetwork]: string } = {
```
Comment (thedoublejay) [inline apps/web/src/config/networkUrl.ts:3]: Can you create a github issue for this to replace, then add a TODO comment so we don't forget

File: apps/web/src/store/defichain.tsx [hunk 3]
```diff
@@ -58,9 +57,8 @@ export const bridgeApi = createApi({
       extraOptions: { maxRetries: 3 },
     }),
     getAddressDetail: builder.mutation<AddressDetails, any>({
-      query: ({ network, address }) => ({
-        url: `/wallet/address/${address}`,
-        params: { network },
+      query: ({ baseUrl, address }) => ({
+        url: `${baseUrl}/wallet/address/${address}`,
```
Comment (thedoublejay) [inline apps/web/src/store/defichain.tsx:61]: I think this will break because we needt o have `defichain` as part of it

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: apps/web/src/config/networkUrl.ts (hunk 2 above)
Comment (kevzzsk) [reply apps/web/src/config/networkUrl.ts:4]: Done #435

**Pull request merged**